mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/PojoMassIndexer.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.massindexing;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.hibernate.search.util.common.annotation.Incubating;
+
+/**
+ * A MassIndexer is useful to rebuild the indexes from the
+ * data contained in the database.
+ * This process is expensive: all indexed entities and their
+ * indexedEmbedded properties are scrolled from database.
+ *
+ * @author Sanne Grinovero
+ * @param <N> the {@link PojoMassIndexer}
+ */
+public interface PojoMassIndexer<N extends PojoMassIndexer<?>> {
```

Comment:
This interface should be SPI.

Also, as far as I can tell, the type argument `N` is not necessary.

Reply:
I'll change it.